x-pack/filebeat/input/awscloudwatch/cloudwatch.go

```diff
@@ -76,23 +76,25 @@ func (p *cloudwatchPoller) run(svc *cloudwatchlogs.Client, logGroup string, star
 func (p *cloudwatchPoller) getLogEventsFromCloudWatch(svc *cloudwatchlogs.Client, logGroup string, startTime int64, endTime int64, logProcessor *logProcessor) error {
 	// construct FilterLogEventsInput
 	filterLogEventsInput := p.constructFilterLogEventsInput(startTime, endTime, logGroup)
+	paginator := cloudwatchlogs.NewFilterLogEventsPaginator(svc, filterLogEventsInput)
+	for paginator.HasMorePages() {
+		filterLogEventsOutput, err := paginator.NextPage(context.TODO())
```

Comment:
here how I implemented pagination error handling on the s3 objects listing:
https://github.com/elastic/beats/pull/31224/files#diff-c6357a5d68a965123a3b3dc3cbf0bddfe7a5628dceb5327885da33f1735a3facR158-R173


it might be too defensive, up to you if using the same pattern for the two paginators in this commit


in the case of s3 `ListObjectsPaginator.NextPage()` an error can be caused by `!ListObjectsPaginator.HasMorePages()`. we should never reach such state since the for loop would have already exited.

if the cwlogs `LogEventsPaginator.NextPage()` and `DescribeLogGroupsPaginator.NextPage()` do not have an error caused by `!HasMorePages()` no need of the extra if